core/material.js

```diff
@@ -1,26 +1,45 @@
 import { create_program_object, create_shader_object, gl } from './context';
 import { Transform, Render } from '../components';
 
+import { vertex } from './shaders';
+import { fragment } from './shaders';
+
 export class Material {
-  constructor() {
+  constructor(options) {
     this.uniforms = {};
     this.attribs = {};
+    this.features = [];
+    this.features_from_components(Array.from(options.components) || []);
```

Comment:
Given that we use `options.components` in the `Entity` constructor as well, do you think there's a naming conflict here? In the `Entity` constructor, `components` is an array of component instances. Here we're expecting component classes. Perhaps rename this to `requires`?